hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/OzoneManagerProtocolClientSideTranslatorPB.java

```diff
@@ -347,7 +355,12 @@ private OMResponse submitRequest(OMRequest omRequest)
 
       return omResponse;
     } catch (ServiceException e) {
-      throw ProtobufHelper.getRemoteException(e);
+      NotLeaderException notLeaderException = getNotLeaderException(e);
+      if (notLeaderException == null) {
+        throw ProtobufHelper.getRemoteException(e);
+      } else {
+        throw new NotLeaderException("Could not determine OM Leader.");
```

Comment:
we can throw notLeaderException which we have received in Line 358.

Reply:
The error message from the previous NotLeaderException points to a single OM node. That is why I am creating a new Exception here to specify that none of the OMs are the leader.